### ISSUE TITLE
Rename confusing fignumber name into itemnumber?

### ISSUE THREAD

**Issue by yves-chevallier** [sphinx-doc/sphinx#6637]
# Issue?
I was fooled by the naming convention `fignumber` that I thought only related to figures. I lately discovered that `fignumber` applies to equations, tables, figures and listings. The terminology is therefore a bit confusing. 

# Solution?
I propose to rename the `fignumber` into something else like `itemnumber` or anything better you guys propose.

**Comment by leotrs** [comment]
Agreed. It would also help to write some documentation about what it does, as I have been writing my own auto-numbering scheme, and after finding fignumber I now believe I have been reinventing the wheel...
